### PR TITLE
infinite loop

### PR DESCRIPTION
**Pull request by wowlocal**
Let len can return 0, it happens when outpusStream is nil. This loop ```while !sOperation.isCancelled)```  will never stop because it will always fall into the else block ```total += len``` where total = 0 and len = 0

### CODE REVIEW

File: Sources/WebSocket.swift [hunk 1]
```diff
@@ -479,7 +479,7 @@ open class WebSocket : NSObject, StreamDelegate, WebSocketClient, WSStreamDelega
      - parameter completion: The (optional) completion handler.
      */
     open func write(string: String, completion: (() -> ())? = nil) {
-        guard isConnected else { return }
+        guard !isConnected else { return }
```
Comment (nuclearace) [inline Sources/WebSocket.swift]: This shouldn't change right?

**Comment by wowlocal** [comment]
Right

File: Sources/WebSocket.swift (hunk 1 above)
Comment (olesDem) [reply Sources/WebSocket.swift]: This is not the root of the problem. It's 

 public func write(data: Data) -> Int {
        guard let outStream = outputStream else {return 0}
        let buffer = UnsafeRawPointer((data as NSData).bytes).assumingMemoryBound(to: UInt8.self)
        return outStream.write(buffer, maxLength: data.count)
    }

Problem

**Comment by daltoniam** [comment]
I manually addressed this in version 3.0.3 (just released), but thank you for the PR.

**Pull request closed without merging**